inductor: enhance conv+binary fusion path test for cpu path

XiaobingSuper opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100058
* #100057

The motivations for this PR are:

1. Add negative/positive testing for conv+binary fusion path.
2. Add alias check for in-place fusion path: if the write buffer is an alias tensor, we will lower the out-place path(one test is also added).
3. Fix https://github.com/pytorch/pytorch/issues/99842.


cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

XiaobingSuper committed:
inductor: enhance conv+binary fusion path test for cpu path

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: enhance conv+binary fusion path test for cpu path"


The motivations for this PR are:

1. Add negative/positive testing for conv+binary fusion path.
2. Add alias check for in-place fusion path: if the write buffer is an alias tensor, we will lower the out-place path(one test is also added).
3. Fix https://github.com/pytorch/pytorch/issues/99842.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: enhance conv+binary fusion path test for cpu path"


The motivations for this PR are:

1. Add negative/positive testing for conv+binary fusion path.
2. Add alias check for in-place fusion path: if the write buffer is an alias tensor, we will lower the out-place path(one test is also added).
3. Fix https://github.com/pytorch/pytorch/issues/99842.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jgong5 commented on torch/_inductor/fx_passes/mkldnn_fusion.py:
```diff
@@ -353,6 +352,49 @@ def fn(match, *args, **kwargs):
 
         return fn
 
+    def _register_binary_unary_inplace_fusion_lowering(
```
```suggestion
    def _register_binary_unary_maybe_inplace_fusion_lowering(
```

XiaobingSuper committed:
Update on "inductor: enhance conv+binary fusion path test for cpu path"


The motivations for this PR are:

1. Add negative/positive testing for conv+binary fusion path.
2. Add alias check for in-place fusion path: if the write buffer is an alias tensor, we will lower the out-place path(one test is also added).
3. Fix https://github.com/pytorch/pytorch/issues/99842.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper commented:
@pytorchbot merge
